src/main/java/io/vertx/core/json/JsonArray.java

```diff
@@ -688,7 +694,11 @@ public boolean equals(Object o) {
 
   @Override
   public int hashCode() {
-    return list.hashCode();
+    int h = 0;
+    for (Object value : this) {
+      h += JsonUtil.hashCode(value);
```

Comment:
How about using a multiplier when combining hashcodes in order to reduce collisions? Same comment for Json objects.

Reply:
I copied what is in JDK

Reply:
good idea